### PR TITLE
#30009 Resolve regression from CSP updates

### LINKED ISSUE #30009: Users can not assign widgets to individual categories/products
### Preconditions (*)

1. Magento 2.4.0
2.

### Steps to reproduce (*)

1. Login to admin
2. Create Widget
3. Assign widget to anchor categories
4. Select "Specific Categories"

### Expected result (*)

1. Chooser appears
2. image

### Actual result (*)

1. Chooser does not appear 
2. image

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mtbottens**
### Description (*)

Resolves regression caused from: https://github.com/magento/magento2/commit/4758b4366c3881ce23272e39ea5f5d5a4417a1d7#diff-64b5862c98e5392810d14d8c4d605cb5

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30009

### Manual testing scenarios (*)

1. Navigate to magento admin
2. Create a widget
3. Add widget to layered category pages
4. Assign to specific categories
5. Observe selector appears

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogyar** [comment]
Hi @mtbottens. Thank you for the fix. According to the [definition of done](https://devdocs.magento.com/contributor-guide/contributing_dod.html) all the changes should be covered by automated tests. I would ask you to cover this case with an MFTF test.  

Thank you!

**Review by sidolov — CHANGES_REQUESTED** [review]
@mtbottens please, cover changes with automated test

**Comment by engcom-Charlie** [comment]
@mtbottens could you please cover your changes by an automated test? Otherwise, we can't proceed with your PR.
Thank you.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by mtbottens** [comment]
@magento run all tests

File: app/code/Magento/Widget/Test/Mftf/Test/AdminContentWidgetsDisplayOnSpecificEntitiesTest.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            <group value="widget"/>
+        </annotations>
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="LoginAsAdmin"/>
+        </before>
+        <after>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+        </after>
+        <amOnPage url="{{AdminNewWidgetPage.url}}" stepKey="createWidgetPage"/>
+        <actionGroup ref="AdminCreateSpecificEntityWidgetActionGroup" stepKey="fillForm">
```
Comment (rogyar) [inline app/code/Magento/Widget/Test/Mftf/Test/AdminContentWidgetsDisplayOnSpecificEntitiesTest.xml:27]: Great job! I would suggest adjusting this test slightly. Ideally, if there's no chooser (original issue) we should have a failing test on an assertion. Wi the current test we will have rather an error (absent selector) than a failing assertion. 

My recommendation is to make the test work in the following way. 
- Select categories radio
- Select container value
- **Assert** that the chooser is present on the page

I hope it makes sense

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
@mtbottens could you please resolve conflict with mainline and look at https://github.com/magento/magento2/pull/30010#discussion_r507166298?
Thank you.

**Comment by engcom-Charlie** [comment]
@mtbottens please resolve conflict with mainline and look at https://github.com/magento/magento2/pull/30010#discussion_r507166298. Otherwise, we can't proceed with your PR.
Thank you.

**Comment by mtbottens** [comment]
@magento run all tests

File: app/code/Magento/Widget/Test/Mftf/Test/AdminContentWidgetsDisplayOnSpecificEntitiesTest.xml (hunk 1 above)
Comment (mtbottens) [reply app/code/Magento/Widget/Test/Mftf/Test/AdminContentWidgetsDisplayOnSpecificEntitiesTest.xml:27]: @rogyar Updated PR per your feedback

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8396](https://jira.corp.magento.com/browse/ENGCOM-8396) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario

1. Go to Content->Widgets
2. Click on Add Widget button
3. Select any widget type and Design theme
4. Click 'Continue' button
5. Scroll down and click on 'Add Layout Update' button
6. Select 'Anchor Categories' from Display on drop down list
7. Select 'Specific Categories' radio button

**Before** applying changes provided in this PR
before

**After** switching to this PR
Category selection

**Comment by gabrieldagama** [comment]
@magento run all tests

**Pull request merged**